internal/vds/axis.h

```diff
@@ -0,0 +1,38 @@
+#ifndef AXIS_H
+#define AXIS_H
+
+#include <memory>
+#include <string>
+
+#include <OpenVDS/OpenVDS.h>
+
+#include "vds.h"
+
+class Axis {
+public:
+    Axis(
+        enum api_axis_name const                axis_name,
+        OpenVDS::VolumeDataLayout const * const layout
+    );
+
+    int get_number_of_points() const noexcept(true);
+
+    int get_min() const noexcept(true);
+    int get_max() const noexcept(true);
+
+    std::string get_unit() const noexcept(true);
+
+    int get_voxel_index() const noexcept(true);
+    int get_index() const noexcept(true);
```

Comment:
I agree that this is not optimal. Regarding the magic 0s and 1s and the lack of use of this function: We could use `get_index()` a bit more nicely here to remove the magic numbers from the `coordinate` variable in `fetch_fence`. This would also be in agreement with the current meaning of the `index` member of an axis. Then the function would be used more often than `get_voxel_index()`.

Reply:
I'm leaning the other way. I think this PR paves the way for some further nice improvements to fence and slice. But that's for another PR. So I would say revert to hardcoding 0s and 1s in fence and remove this function, _for now_